File: php/testcode/benchmark_test_00012.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest00012(BenchmarkRequest $req): BenchmarkResponse {
    $input = ['lang' => $req->param('lang'), 'theme' => $req->param('theme')];
    extract($input, EXTR_PREFIX_ALL, 'usr_');
    $lang = $usr_lang ?? 'en';
    return BenchmarkResponse::ok("lang=$lang");
}

File: rust/testcode/benchmark_test_00012.rs
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let filename = req.param("filename");
    let content = req.param("content");
    let basename = std::path::Path::new(&filename)
        .file_name()
        .and_then(|n| n.to_str())
        .unwrap_or("upload");
    let _ = std::fs::write(format!("/uploads/{}", basename), content.as_bytes());
    super::shared::BenchmarkResponse::ok("Saved")
}
